tomcat/src/main/java/org/apache/coyote/publisher/InputStreamRequestPublisher.java

```diff
@@ -0,0 +1,103 @@
+package org.apache.coyote.publisher;
+
+import org.apache.coyote.common.Headers;
+import org.apache.coyote.common.MessageBody;
+import org.apache.coyote.exception.CoyoteIOException;
+import org.apache.coyote.request.HttpRequest;
+import org.apache.coyote.request.RequestLine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class InputStreamRequestPublisher {
+
+    private static final Logger log = LoggerFactory.getLogger(InputStreamRequestPublisher.class);
+
+    private static final String HEADER_END_CONDITION = "";
+
+    private final RequestLine requestLine;
+    private final Headers headers;
+    private final MessageBody messageBody;
+
+    private InputStreamRequestPublisher(final RequestLine requestLine,
+                                        final Headers headers,
+                                        final MessageBody messageBody
+    ) {
+        this.requestLine = requestLine;
+        this.headers = headers;
+        this.messageBody = messageBody;
+    }
+
+    public static InputStreamRequestPublisher read(final InputStream inputStream) {
+        try {
+            final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+            final RequestLine requestLine = RequestLinePublisher
+                    .read(readRequestLine(reader))
+                    .toRequestLine();
+
+            final Headers headers = RequestHeaderPublisher
+                    .read(readHeaderValues(reader))
+                    .toHeaders();
+
+            if (isContentLengthIsNullOrWrong(headers)) {
+                return createEmptyMessageBody(requestLine, headers);
+            }
+            return createWithMessageBody(reader, requestLine, headers);
```

Comment:
바디도 위 line, header와 동일하게 하면 더 가독성이 좋아질 수..도?

```suggestion
final MessageBody messageBody = createMessageBody(reader, requestLine, headers); // 내부에서 분기처리
return new InputStreamRequestPublisher(requestLine, headers, messageBody);
```